pymc_marketing/mmm/media_transformation.py

```diff
@@ -0,0 +1,278 @@
+@dataclass
+class MediaTransformation:
+    """Wrapper for applying adstock and saturation transformation to media data.
+
+    Parameters
+    ----------
+    adstock : AdstockTransformation
+        The adstock transformation to apply.
+    saturation : SaturationTransformation
+        The saturation transformation to apply.
+    adstock_first : bool
+        Flag to apply the adstock transformation first.
+
+    Attributes
+    ----------
+    first : AdstockTransformation | SaturationTransformation
+        The first transformation to apply.
+    second : AdstockTransformation | SaturationTransformation
+        The second transformation to apply.
+
+    """
+
+    adstock: AdstockTransformation
+    saturation: SaturationTransformation
+    adstock_first: bool
+
+    def __post_init__(self):
+        """Set the first and second transformations based on the adstock_first flag."""
+        self.first, self.second = (
+            (self.adstock, self.saturation)
+            if self.adstock_first
+            else (self.saturation, self.adstock)
+        )
+
+    def __call__(self, x, dim):
+        """Apply adstock and saturation transformation to media data.
+
+        Parameters
+        ----------
+        x : pt.TensorLike
+            The media data to transform.
+        dim : str
+            The dimension of the parameters.
+
+        Returns
+        -------
+        pt.TensorVariable
+            The transformed media data.
+
+        Examples
+        --------
+        Apply the media transformation to media data:
+
+        .. code-block:: python
+
+            from pymc_marketing.mmm import (
+                GeometricAdstock,
+                HillSaturation,
+                MediaTransformation,
+            )
+
+            media_data = ...
+
+            media_transformation = MediaTransformation(
+                adstock=GeometricAdstock(l_max=15),
+                saturation=HillSaturation(),
+                adstock_first=True,
+            )
+
+            coords = {
+                "date": ...,
+                "media": ...,
+            }
+            with pm.Model(coords=coords) as model:
+                transformed_media_data = media_transformation(
+                    media_data,
+                    dim="media",
+                )
+
+        """
+        return self.second.apply(self.first.apply(x, dim), dim)
+
+
+@dataclass
+class MediaConfig:
+    """Configuration for a media transformation to certain media channels.
+
+    Parameters
+    ----------
+    name : str
+        The name of the media transformation and prefix of all media variables.
+    columns : list[str]
+        The media channels to apply the transformation to.
+    media_transformation : MediaTransformation
+        The media transformation to apply to the media channels.
+
+    """
+
+    name: str
+    columns: list[str]
+    media_transformation: MediaTransformation
+
+
+MediaConfigs = list[MediaConfig]
+
+
+def apply_media_transformation(
```

Comment:
If these are wrapped in a class, then it should be pretty easy to make a consistent API with the `apply` method. 

WDYT?

Reply:
yes!